base/dict.jl

```diff
@@ -691,6 +691,12 @@ end
 
 filter!(f, d::Dict) = filter_in_one_pass!(f, d)
 
+function reduce(::typeof(merge), items::Vector{<:Dict})
```

Comment:
Any reason not to allow `AbstractVector` ?
(sorry for a last minute comment!)

Reply:
IIRC it triggered ambiguities (that's probably buried in a comment above). It would be useful to try to generalize, but for now I'd say it's already an improvement.